src/local/base.py

```diff
@@ -0,0 +1,130 @@
+import asyncio
+from contextlib import suppress
+import enum
+import glob
+from logging import getLogger
+import os
+import subprocess
+from typing import Iterable, Optional
+import webbrowser
+from attr import dataclass
+
+from galaxy.api.types import LocalGame
+
+from .shared import load_vdf
+
+log = getLogger(__name__)
+
+
+class StateFlags(enum.Flag):
+    """StateFlags from appmanifest.acf file"""
+    Invalid = 0
+    Uninstalled = 1
+    UpdateRequired = 2
+    FullyInstalled = 4
+    Encrypted = 8
+    Locked = 16
+    FilesMissing = 32
+    AppRunning = 64
+    FilesCorrupt = 128
+    UpdateRunning = 256
+    UpdatePaused = 512
+    UpdateStarted = 1024
+    Uninstalling = 2048
+    BackupRunning = 4096
+    Reconfiguring = 65536
+    Validating = 131072
+    AddingFiles = 262144
+    Preallocating = 524288
+    Downloading = 1048576
+    Staging = 2097152
+    Committing = 4194304
+    UpdateStopping = 8388608
+
+
+@dataclass
+class Manifest:
+    path: str
+    
+    def id(self):
+        return os.path.basename(self.path)[len('appmanifest_'):-4]
+    
+    def app_size(self):
+        with load_vdf(self.path) as manifest:
+            app_state = manifest["AppState"]
+            if StateFlags.FullyInstalled in StateFlags(int(app_state["StateFlags"])):
+                return int(app_state["SizeOnDisk"])
+            else:
+                return int(app_state["BytesDownloaded"])
+
+
+class BaseClient:
```

Comment:
Consider inheriting ABC.abc (abstract base class) seeing as the MacOS and windows both inherit them. you can then decorate the functions with `@abstractmethod` and simply `pass` instead of `raise NotImplementedError`. We'll get nice(ish) runtime checks if the abstract classes don't override them. Python 3.12 is adding an `@override` annotation which would really help with typechecking but it's not here yet. Another reason i prefer C#/C++/Java, lol.

Reply:
could include the overrides module, and seeing as i'm hoping to make the caches implement an abstract parent, it might actually be useful, but for now i'm holding off

Reply:
yeah i was thinking about both but `raise NotImplementedError` saved a line of code per method (either way a meaningful error will be raised if not overriden). overrides module could be useful but yeah lets see if we can do it without. i'll make sure to change it to use `abc` 👍